libs/atqr/domain/src/lib/domain-spec/stories/visitorAdoptsGoal.disl

```diff
@@ -0,0 +1,28 @@
+read_also api.disl
+read_also enums.disl
+read_also ui.disl
+
+entity Goal
+  @goal: string
+  @challenges: Challenge[]
+  @pursuers: returns @challenges.map(c => c.enrollments.map(e => e.player)).flat()
+
+root Challenge
+  CreateChallenge?
+  AdoptGoal? :email :goal
+
+  rules:
+    /**
+    * visitor's email must be validated as a player pursuing or not this goal through Goal's @pursuer
+    */
+    when :email not in @pursuer's email
+      CreateChallenge?
```

Comment:
This would be a method call. Commands represent user requests to the system. Also, `createChallenge` takes a lot more arguments

Reply:
I changed it to a method call, but kept the two properties below it cause they are the only ones I need to be already guarding values. I imagine the rest of them are already being handled by the createChallenge method.

Reply:
There's no "imagine" allowed in coding :) `createChallenge` is a method and can only handle what you pass to it